barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.cpp

```diff
@@ -125,10 +119,8 @@ void ClientIVC::precompute_folding_verification_keys()
 
     vks.kernel_vk = std::make_shared<VerificationKey>(prover_instance->proving_key);
 
-    // Clean the ivc state
-    goblin.op_queue = std::make_shared<Goblin::OpQueue>();
-    goblin.merge_proof_exists = false;
-    GoblinMockCircuits::perform_op_queue_interactions_for_mock_first_circuit(goblin.op_queue);
+    // Clean the Goblin state (reinitialise op_queue with mocking and clear merge proofs)
+    goblin = Goblin();
```

Comment:
This seems a bit weird to me. I know it's not being introduced in this PR but is there some reason why we don't simply instantiate a new goblin instance for use within the scope of this function?

Reply:
that is a good point, will aim to do that, thanks!

Reply:
added to do to aim to move vk precomputaion out of client ivc